sisl/viz/plotly/plots/bands.py

```diff
@@ -0,0 +1,693 @@
+from functools import partial
+import itertools
+
+import numpy as np
+import xarray as xr
+import plotly.express as px
+
+import sisl
+from ..plot import Plot, PLOTS_CONSTANTS, entry_point
+from ..plotutils import find_files
+from ..input_fields import TextInput, SwitchInput, ColorPicker, DropdownInput,\
+     IntegerInput, FloatInput, RangeInput, RangeSlider, QueriesInput, ProgramaticInput, FunctionInput, SileInput, \
+         PlotableInput, SpinSelect
+from ..input_fields.range import ErangeInput
+
+
```

Comment:
when running the tutorial I would have liked to be able to change the limits interactively, can this easily be done in the `Bands plot`? ;)

Reply:
Do you mean an input field that updates the plot automatically when it is changed? 

In the GUI you have the input fields, but you need to click update settings to see the effect (although there's also the shortcut shift+enter). I also thought that some settings are quick enough to be made interactively. In fact, I opened an issue for this: https://github.com/pfebrer96/sisl/issues/29.

I still haven't defined how input fields should show up in a jupyter notebook, so unfortunately in that case you need to `plot.update_settings()`. But it is one of the things I wanted to do. It should be quite easy as jupyter widgets already exist for this. The only thing I have to thing about is how to display them exactly. For example, I think by default they should be hidden, with a little tab to expand them when clicked.

Reply:
Yeah, that sounds like it could solve the problem. It was just that if you have a *bands* file with -10 eV to 100 eV, you basically can't use the plot for anything ;)

Reply:
Maybe I didn't understand the question. What did you mean by "interactively change"?

Because you can definitely change the settings of the plot, like:

```python
from sisl.viz import Plot

bands_plot = Plot("path/to/bands/file", Erange=[-20, 30])

bands_plot.update_settings(Erange=[-5, 10])
# or
bands_plot.update_settings(bands_range=[40, 76])
```

Did you mean this?

Reply:
Yes, this seems like it would be quite useful to change on the fly ;)

Exactly this.

Reply:
> Yes, this seems like it would be quite useful to change on the fly ;)
> 
> Exactly this.

Also, for the future.

I think that plots should have *common* keywords for simple things, e.g. xmin/xmax, ymin/ymax. The reason is that users will typically guess this (due to matplotlib) and it would be annoying if keywords are different depending on the plot.

Reply:
> Yes, this seems like it would be quite useful to change on the fly ;)

You can change any setting with `plot.update_settings`. You can see all the current settings by doing `print(plot)` or `plot.settings` (which is a dict). Or they are also in the help message of `plot.update_settings()`. So doing shift+tab inside the function call will make them appear. The plot will rerun only the parts that need to be executed when that setting in particular changes.

> Also, for the future.
> 
> I think that plots should have common keywords for simple things, e.g. xmin/xmax, ymin/ymax. The reason is that users will typically guess this (due to matplotlib) and it would be annoying if keywords are different depending on the plot.

Yes, this is why I create specific input fields that should be used, because then I enforce consistency. For example `ErangeInput` is the input field that should be used by an energy range, and it will always be called "Erange". (it may also implement useful things regarding energy ranges in the future.)

The layout attributes (xmin/xmax, ymin/ymax) are not really handled by the settings, but directly by plotly. So they are as consistent as plotly makes them and not specific to the visualization module (one can refer to plotly's documentation). They are updated with `plot.update_layout()`.

E.g.:
```python
from sisl.viz import Plot

bands_plot = Plot("path/to/bands/file", Erange=[-20, 30])

# Update the bands that are put into the plot
bands_plot.update_settings(Erange=[-5, 10])

# Update the range of the y axis (which is a separate thing)
# and is handled by plotly.
bands_plot.update_layout(yaxis={"range": [-3, 30]})
# or
bands_plot.update_layout(yaxis_range=[-3, 30])
# or
bands_plot.layout.yaxis.range = [-3, 30]
# or
bands_plot.layout.yaxis = {"range": [-3, 30]}
```